src/Module/Slim.php

```diff
@@ -38,44 +46,103 @@
  * actor: FunctionalTester
  * modules:
  *   enabled:
- *     - DoclerLabs\CodeceptionSlimModule\Module\Slim:
- *         application: 'app/bootstrap.php'
  *     - REST:
  *         depends: DoclerLabs\CodeceptionSlimModule\Module\Slim
+ *
+ *   config:
+ *     DoclerLabs\CodeceptionSlimModule\Module\Slim:
+ *       application: 'app/bootstrap.php'
  * ```
  */
 class Slim extends Framework
 {
     /** @var App */
     public $app;
 
+    /** @var string[] */
+    protected $config = [
+        'connector' => SlimPsr7::class,
+    ];
+
     /** @var array */
     protected $requiredFields = ['application'];
 
     /** @var string */
     private $applicationPath;
 
+    /** @var string */
+    private $connectorClass;
+
     public function _initialize(): void
     {
-        $this->applicationPath = Configuration::projectDir() . $this->config['application'];
+        $applicationPath = Configuration::projectDir() . $this->config['application'];
+        if (!file_exists($applicationPath)) {
+            throw new ModuleConfigException(
+                static::class,
+                sprintf(
+                    "\nApplication file doesn't exist.\nPlease, check path for php file: `%s`",
+                    $applicationPath
+                )
+            );
+        }
 
-        if (!file_exists($this->applicationPath)) {
+        $connectorClass = $this->config['connector'];
+        if (!class_exists($connectorClass)) {
             throw new ModuleConfigException(
                 static::class,
-                "\nApplication file doesn't exist.\n"
-                . 'Please, check path for php file: ' . $this->applicationPath
+                sprintf(
+                    "\nUnable to load `%s` connector.\nPlease, check connector configuration.",
+                    $connectorClass
+                )
             );
         }
 
+        $this->applicationPath = $applicationPath;
+        $this->connectorClass  = $connectorClass;
+
         parent::_initialize();
     }
 
     public function _before(TestInterface $test): void
     {
+        /* @noinspection PhpIncludeInspection */
         $this->app = require $this->applicationPath;
 
-        $this->client = new SlimConnector();
-        $this->client->setApp($this->app);
+        // Check if app instance is ready.
+        if (!$this->app instanceof App) {
+            throw new ConfigurationException(
+                sprintf(
+                    "\n  Unable to bootstrap slim application.\n  Application file must return with `%s` instance.",
+                    App::class
+                )
+            );
+        }
+
+        // Check connector requirements.
+        $connector = new $this->connectorClass();
+        if (!$connector instanceof AbstractBrowser) {
```

Comment:
Right now, I have a vision, that we don't need these 2 instances checks, as it effectively reduces the class to the single one existing SlimPsr7 connector.

Do you believe that someone else will create custom connectors for Slim4 and codeception? My bet is "no" ) 
Thus, either we don't need this flexibility at all (to remove this connectorClass setting) or remove extra checks here and wait for our class only.

Reply:
I don't expect any contribution, but this is about the possibility of extension. Removed now, since there is no such requirements, we can extend later if necessary.